crates/matrix-sdk-ui/src/timeline/inner/mod.rs

```diff
@@ -915,7 +915,52 @@ impl<P: RoomDataProvider> TimelineInner<P> {
             }
         }
 
-        trace!("Done updating sender profiles");
+        trace!("Done updating missing sender profiles");
+    }
+
+    /// Update the profiles of the given senders, even if they are ready.
+    pub(super) async fn force_update_sender_profiles(&self, sender_ids: &BTreeSet<&UserId>) {
+        trace!("Forcing update of sender profiles: {sender_ids:?}");
+
+        let mut state = self.state.write().await;
+        let mut entries = state.items.entries();
+        while let Some(mut entry) = entries.next() {
```

Comment:
Why not:

```rust
for mut entry in entries { … }
```

?

Reply:
[`ObservableVectorEntries`](https://docs.rs/eyeball-im/latest/eyeball_im/struct.ObservableVectorEntries.html) does not implement `Iterator`, it just has a `.next()` method.